sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -35,20 +41,38 @@ public EventGridClient(Uri endpoint, AzureKeyCredential credential)
         {
         }
 
-        /// <summary>Initalizes an instance of EventGridClient</summary>
+        /// <summary>Initalizes an instance of the<see cref="EventGridClient"/> class</summary>
         /// <param name="endpoint">topic endpoint</param>
         /// <param name="credential">used to connect to Azure</param>
         /// <param name="options">configuring options</param>
         public EventGridClient(Uri endpoint, AzureKeyCredential credential, EventGridClientOptions options)
         {
             Argument.AssertNotNull(credential, nameof(credential));
             options ??= new EventGridClientOptions();
-            _hostName = endpoint.Host;
+            _apiVersion = options.GetVersionString();
+            _endpoint = endpoint;
+            _key = credential;
             HttpPipeline pipeline = HttpPipelineBuilder.Build(options, new AzureKeyCredentialPolicy(credential, Constants.SasKeyName));
             _serviceRestClient = new ServiceRestClient(new ClientDiagnostics(options), pipeline, options.GetVersionString());
             _clientDiagnostics = new ClientDiagnostics(options);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventGridClient"/> class.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        /// <param name="options"></param>
+        public EventGridClient(Uri endpoint, SharedAccessSignatureCredential credential, EventGridClientOptions options = default)
```

Comment:
nit: we should follow the same pattern as for AzureKeyCredential ctor where we have separate ctors, one having options and one not having it.